### DISCUSSION TITLE
Way to transform values after onSubmit?

### DISCUSSION THREAD

**Discussion by go-mad** [react-hook-form/react-hook-form#3202] [Q&A] [answered]
Is there any way to transform an input value to something else when onSubmit triggers? 

For example, I have a `<Dropdown/>` component which sets the value as `{label: "Europe", value: "EU"}`. But when I submit, the final data which I need is only "EU". 

I'm using the register method to register the field name `register("country", {label: "Europe", value: "EU"})`. When I submit I get `{country: {label: "Europe", value: "EU"}}` as the onSubmit data, but I want it as `{country: "EU"}`

Any way we can write a transform function to do the same, but I was just wondering whether it's possible using RHF itself.

**Comment by bluebill1049** [comment, marked as answer]
Not sure if those will help: 

- https://react-hook-form.com/advanced-usage#ExtendController
- https://codesandbox.io/s/react-hook-form-parse-and-format-textarea-furtc

**Reply by go-mad** [reply]
The links you shared is using the `<Controller />` component. I'm using the `register` method directly. So in my case, it won't help.

**Reply by bluebill1049** [reply]
there is no transform method at `register`, if you need to transform use Controller.

**Reply by Noitidart** [reply]
Is this still true today with v7? I tried this but it acted weird and made my field controlled weirdly. I don't want to parse it back, as its uncontrolled, i just want to transform on the way out:


```
  const { onChange, ...registration } = props.form.register('foo');

  const handleChange = e => {
    const value = e.target.value;
    onChange({
      ...e,
      target: {
        ...e.target,
        value: value.trim() === '' ? undefiend : parseInt(value, 10)
      }
    });
  };

return <input {...registration} onChange={handleChange} />
```

**Reply by bluebill1049** [reply]
Use valueAs?

**Reply by Noitidart** [reply]
Ah thank you. That works cool!

**Comment by hanza93** [comment]
> Is there any way to transform an input value to something else when onSubmit triggers?
> 
> For example, I have a `<Dropdown/>` component which sets the value as `{label: "Europe", value: "EU"}`. But when I submit, the final data which I need is only "EU".
> 
> I'm using the register method to register the field name `register("country", {label: "Europe", value: "EU"})`. When I submit I get `{country: {label: "Europe", value: "EU"}}` as the onSubmit data, but I want it as `{country: "EU"}`
> 
> Any way we can write a transform function to do the same, but I was just wondering whether it's possible using RHF itself.

I'm using react-select and I'm having the same issue, any idea how to solve? @go-mad
